Fabric.Identity.API/scripts/tests/identity-cli-tests.ps1

```diff
@@ -0,0 +1,59 @@
+# Force re-import to pick up latest changes
+Import-Module ..\identity-cli.psm1 -Force
+
+describe 'Get-AccessToken' {
+    Context 'Unit Tests' {
+        Context 'Valid Request' {
+            It 'Should return an access token when valid response' {                
+                $mockUrl = New-MockObject -Type Uri
+                Mock -ModuleName identity-cli -CommandName Invoke-RestMethod { return @{
+                    access_token = "return token"
+                    }
+                }
+                
+                $response = Get-AccessToken -identityUrl $mockUrl -clientId "id" -secret "secret" -scope "fabric/identity.manageresources"
+                $response | Should -Be "return token"
+            }
+        }
+
+        Context 'Invalid request' {    
+            It 'Should return the error when invalid response' {
+                Mock -ModuleName identity-cli -CommandName Invoke-RestMethod { throw  New-Object -TypeName "System.Net.WebException" -ArgumentList "Exception" }
+                
+                {Get-AccessToken -identityUrl "url" -clientId "id" -secret "secret" -scope "fabric/identity.manageresources"} | Should -Throw "Exception" -ExceptionType System.Net.WebException
+            }
+        }
+    }
+}
+
+describe 'Get-FabricInstallerAccessToken' {
+    Context 'Unit Tests' {
+        Context 'Valid Request' {
+            It 'Should return an access token when valid response' {
+                $mockUrl = New-MockObject -Type Uri
+                Mock -ModuleName identity-cli -CommandName Invoke-RestMethod { return @{
+                    access_token = "return token"
+                    }
+                }
+
+                $response = Get-FabricInstallerAccessToken -identityUrl $mockUrl -secret "Secret"
+
+                $response | Should -Be "return token"
+            }
+        }
+
+        Context 'Invalid Request' {
+            It 'Should return an exception when invalid response' {
+                Mock -ModuleName identity-cli -CommandName Invoke-RestMethod { throw  New-Object -TypeName "System.Net.WebException" -ArgumentList "Error" }
```

Comment:
Couldn't find a to create an object outside of the mock setup, trying to set '$exc = New-Object -TypeName "System.Net.WebException" -ArgumentList "Error"' and throwing the variable here raised a different exception that was expected. 
Trying to instead return an object using the same method would return null instead of the object